WordPress/gutenberg
# List Block: Fix numbering style to be applied correctly

Fixes: #52826
Fixes: #54384

**Update:** This PR has changed its approach based on the discussion. For the latest implementation details, please check the comments [here](https://github.com/WordPress/gutenberg/pull/53301#issuecomment-1666753766).

-------

## What?

This PR adds explicit selectors and styles to ensure that numbering styles are applied correctly under certain conditions.

## Why?

From what I have found in [this comment](https://github.com/WordPress/gutenberg/issues/52826#issuecomment-1645582401), this problem is because the WP-Admin `common.css` and `reset.css` applied to the editor are overriding the styles. I also found that these styles are also loaded into the iframe editor in the case of the classic theme.

However, if, for example, custom fields are enabled or a block with apiVersion 2 is present, the editor is not iframed.
As a result, the WP-Admin reset style is applied even in the case of a block theme, which causes problems.

## How?

This issue should be approached on the assumption that a WP-Admin reset style exists.

I first wrote the following selector:

```css
ol {
	&[type="A"] {
		list-style-type: upper-alpha;
	}
	&[type="a"] {
		list-style-type: lower-alpha;
	}
	&[type="I"] {
		list-style-type: upper-roman;
	}
	&[type="i"] {
		list-style-type: lower-roman;
	}
}
```

However, it did not take into account whether the value was uppercase or lowercase, as seen in the screenshot below (confirmed in Chrome and FireFox)

![type](https://github.com/WordPress/gutenberg/assets/54422211/8fe67218-5071-418e-b86c-5cdfbcd51f68)

Therefore, I explicitly generated class names like `is-type-${type}` and applied styles to them.

## Testing Instructions

- Activate custom fields from the Preferences menu.
- Insert a List block and apply the Ordered style.
- Change the Numbering Style.
- (trunk): The numbering style should not change.
- (This PR): The numbering style should change correctly.
